Cockpit 237 (#378)
cockpit-project/cockpit-project.github.io: _posts/2021-02-03-cockpit-237.md

---
title: Cockpit 237
author: marusak
date: 2021-02-03
tags: cockpit linux
slug: cockpit-237
category: release
summary: Cockpit with restyled updates page
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.

Here are the release notes from Cockpit version 237.

> ## Updates: Restyling as preparation for upcoming new fatures

Reviewer: ```suggestion
## Updates: Restyling as preparation for upcoming features
```

> Updates page went through some design polishing as preparation for new upcoming features as support

Reviewer: ```suggestion
The updates page went through some design polishing as preparation for upcoming features such as support
```

> of [tracer](https://tracer.readthedocs.io/en/latest/) and [kpatch](https://github.com/dynup/kpatch).

Reviewer 2: Let's please not announce vaporware.  I'd stop after "for upcoming features.". If you do want to keep it, "features, like support for tracer and kpatch".

... screenshot ... (TODO we need to fix at least the page filling and possibly also dropping title panel).

## Shell: SSH connections to remote machines are only opened when necessary

> Cockpit remembers all remote hosts of a session and used to immediately connect to all of them

Reviewer: ```suggestion
Cockpit remembers all remote hosts of a session and would previously connect to all of them
```

> machine for the first time to restricts possible security issues, as connected host can
> interacted with each other.

Reviewer 2: Runaway sentence, and some grammar issues.

```suggestion
machine for the first time. This restricts possible security issues, as connected hosts can
interact with each other.
```

> when starting the next session. Now a SSH connection is only opened when navigating to the remote

Reviewer: ```suggestion
when starting the next session. Now an SSH connection is only opened when navigating to the remote
```

> machine for the first time.

Reviewer 2: We should give a rationale here: It restricts possible security issues, as any connected host can do *anything* to any other connected host. (With better verbiage, obviously)

## Internal CI improvements

> Team has been focusing in the last two weeks heavily of stabilizing and improving our CI. These

Reviewer 2: In the last two weeks, the team has been focusing heavily on stabilizing.."

> improvements include work on self-testing CI infrastructure as well as implementation of [error
> budget](https://sre.google/sre-book/embracing-risk/). These changes don't have a user-visible effect,

Reviewer 2: ```suggestion
improvements include work on self-testing CI infrastructure as well as [defining](https://github.com/cockpit-project/cockpit/wiki/DevelopmentPrinciples#our-testsci-error-budget) and [implementing](https://images-frontdoor.apps.ocp.ci.centos.org/tests.html) an [error budget](https://sre.google/sre-book/embracing-risk/). These changes don't have a user-visible effect,
```

but will make Cockpit easier to develop in the future. Also we continued working on tests and build system improvements.

> On top all of that there is a bug fix in websocket thanks to a contribution from Asad Mehmood. _Thanks, mehmooda!_

Reviewer 2: ```suggestion
On top all of that there is a bug fix in websocket; _thanks to Asad Mehmood for that contribution!_
```

> On top all of that there is a bug fix in websocket thanks to contribution from mehmooda. _Thanks, mehmooda!_

Reviewer 2: We don't usually mention bug fixes, but if we do, there should be some details about it.

Author: Right, we just wanted to mention external contributor. @allisonkarlitskaya

Reviewer: Maybe include full name?  It's listed on the GitHub profile.

> On top all of that there is a bug fix in websocket thanks to contribution from mehmooda. _Thanks, mehmooda!_

Reviewer: ```suggestion
On top all of that there is a bug fix in websocket thanks to a contribution from mehmooda. _Thanks, mehmooda!_
```
